fl4health/model_bases/autoencoders_base.py

```diff
@@ -0,0 +1,162 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Tuple
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class AutoEncoderType(Enum):
+    BASIC_AE = "BASIC_AE"
+    VARIATIONAL_AE = "VARIATIONAL_AE"
+    CONDITIONAL_VAE = "CONDITIONAL_VAE"
+
+
+class AbstractAE(nn.Module, ABC):
```

Comment:
`AE` throughout this file for class names should be `Ae` by the naming convention that we've unified on for class names with acronyms 🙂